.github/workflows/flatpak-test.yml

```diff
@@ -32,7 +32,7 @@ jobs:
       - name: Smoke-test the installed flatpak
         run: |
           . /etc/profile.d/flatpak.sh
-          xvfb-run dbus-run-session sh -ec '
-            containers/flatpak/test/test-ssh
-            containers/flatpak/test/test-browser
+          xvfb-run sh -ec '
+            dbus-run-session containers/flatpak/test/test-ssh
+            dbus-run-session containers/flatpak/test/test-browser
```

Comment:
I expected a bit more compartmentalisation here: we run several browser tests in `test-browser`, no?  And the compartmentalisation between them is working properly, apparently.

Maybe the problem is that we don't do the waiting before the first test in test-browser (or after the test in test-ssh)?  It seems a bit arbitrary to split it up like this...

On the other hand, if it works...

Reply:
It does work, see the [stress test above](https://github.com/cockpit-project/cockpit/pull/17149#issuecomment-1072131094).

I wouldn't know what exactly to wait for at the start of test-browser (or the end of test-ssh, doesn't matter much). We already added the `flatpak ps` wait loop, and process wait loop. We could probably introspect D-Bus for a bit, but that's more time than I'd like to spend on this.

An alternative might be to simply swap the order of test-browser and test-ssh, and pretend that it's nicer to sort them alphabetically :grin: If you'd like me to test that, I can do that.

Reply:
I think I may have misunderstood what the browser tests are doing.  You're using one instance for all of the tests after all.

In that case, let's just go with your idea here.